reana: create example workflow and run reco on 2011 sample

Following the test in #5  for some CMS 2011 RAW sample, create `reana.yaml` and run the process through REANA using CVMFS.  Compare the obtained AOD resulting from VM run and from REANA run.

@clelange Thanks! interesting project structure; I am now able to run the [demoanalyzer.py](https://github.com/dprelipcean/reana-demo-cms-reco/blob/reana_example_2011/demoanalyzer_cfg.py) inside the CMS VM, but not in reana using [this workflow](https://github.com/dprelipcean/reana-demo-cms-reco/blob/reana_example_2011/reana-analyzer.yaml)

The error I get is:

```console
 fatal error: boost/shared_ptr.hpp: No such file or directory
```
Do you think it's possbile that the `cmsopendata/cmssw_5_3_32` docker image does not contain all dependencies/headers?

Full stack trace:
```console
WARNING: In non-interactive mode release checks e.g. deprecated releases, production architectures are disabled.
>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Entering Package CMSreco/validation
>> Creating project symlinks
Entering library rule at CMSreco/validation
>> Compiling  /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/efd42bab-90c3-4308-ae84-73fb31e63960/CMSSW_5_3_32/src/CMSreco/validation/src/PhysicsObjectsHistos.cc 
In file included from /opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/FWCore/Framework/interface/Frameworkfwd.h:10:0,
                 from /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/efd42bab-90c3-4308-ae84-73fb31e63960/CMSSW_5_3_32/src/CMSreco/validation/src/PhysicsObjectsHistos.cc:18:
/opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/DataFormats/Common/interface/EDProductfwd.h:9:32: fatal error: boost/shared_ptr.hpp: No such file or directory
compilation terminated.
In file included from /opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/FWCore/Framework/interface/Frameworkfwd.h:10:0,
                 from /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/efd42bab-90c3-4308-ae84-73fb31e63960/CMSSW_5_3_32/src/CMSreco/validation/src/PhysicsObjectsHistos.cc:18:
/opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/DataFormats/Common/interface/EDProductfwd.h:9:32: fatal error: boost/shared_ptr.hpp: No such file or directory
compilation terminated.
gmake: *** [tmp/slc6_amd64_gcc472/src/CMSreco/validation/src/CMSrecovalidation/PhysicsObjectsHistos.o] Error 1
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

Hei @clelange, thanks a lot for all the quick feedback!
I had a look at the `boost` libraries both in the VM and in the image, and there was no difference (apart from base paths).

Fooling around a bit and looking at the [cms validation](https://github.com/cms-legacydata-validation/RAWToAODValidation/tree/2011), I noticed that one actually needs not only 2, but 3 level subdirectories in the image, i.e.:
```console
~/CMSSW_5_3_32/src/WorkDir/Reconstruction/Validation $ scram b
>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Nothing to be done for src/WorkDir/Reconstruction/Validation
>> Local Products Rules ..... started
>> Local Products Rules ..... done
gmake[1]: Entering directory `/home/cmsusr/CMSSW_5_3_32'
>> Creating project symlinks
>> Done python_symlink
>> Compiling python modules python
>> All python modules compiled
>> Pluging of all type refreshed.
>> Done generating edm plugin poisoned information
gmake[1]: Leaving directory `/home/cmsusr/CMSSW_5_3_32'
```
whereas:
```console
~/CMSSW_5_3_32/src/WorkDir/Reconstruction $ scram b
Reading cached build data
>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Entering Package WorkDir/Reconstruction
>> Creating project symlinks
Entering library rule at WorkDir/Reconstruction
>> Compiling  /home/cmsusr/CMSSW_5_3_32/src/WorkDir/Reconstruction/src/PhysicsObjectsHistos.cc 
In file included from /opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/FWCore/Framework/interface/Frameworkfwd.h:10:0,
                 from /home/cmsusr/CMSSW_5_3_32/src/WorkDir/reana-demo-cms-reco/src/PhysicsObjectsHistos.cc:18:
/opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/DataFormats/Common/interface/EDProductfwd.h:9:32: fatal error: boost/shared_ptr.hpp: No such file or directory
compilation terminated.
In file included from /opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/FWCore/Framework/interface/Frameworkfwd.h:10:0,
                 from /home/cmsusr/CMSSW_5_3_32/src/WorkDir/Reconstruction/src/PhysicsObjectsHistos.cc:18:
/opt/cms/slc6_amd64_gcc472/cms/cmssw/CMSSW_5_3_32/src/DataFormats/Common/interface/EDProductfwd.h:9:32: fatal error: boost/shared_ptr.hpp: No such file or directory
compilation terminated.
gmake: *** [tmp/slc6_amd64_gcc472/src/WorkDir/Reconstruction/src/WorkDirReconstruction/PhysicsObjectsHistos.o] Error 1
```
Previously, in the VM, I needed only 2.

Or is this
```console
>> Nothing to be done for src/WorkDir/Reconstruction/Validation
```
 what you meant by 

> Otherwise, you will see that scram basically does nothing.

--- a/BuildFile.xml
+++ b/BuildFile.xml
@@ -1,0 +1,11 @@
+<buildfile>
+<use name="FWCore/Framework"/>
+<use name="FWCore/PluginManager"/>
+<use name="FWCore/ParameterSet"/>
+<use name="DataFormats/TrackReco"/>
+<use name="CommonTools/UtilAlgos"/>
+<use name="PhysicsTools/UtilAlgos"/>
+<use name="DataFormats/MuonReco"/>
+<use name="RecoMuon/TrackingTools"/>
+<flags EDM_PLUGIN="1"/>
+</buildfile>

--- a/README.rst
+++ b/README.rst
@@ -2,3 +2,223 @@
  REANA example - CMS reconstruction
 ====================================
 
+.. image:: https://badges.gitter.im/Join%20Chat.svg
+   :target: https://gitter.im/reanahub/reana?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge
+
+.. image:: https://img.shields.io/github/license/reanahub/reana-demo-cms-reco.svg
+   :target: https://raw.githubusercontent.com/reanahub/reana-demo-cms-reco/master/LICENSE
+
+
+About
+======
+This REANA reproducible analysis example demonstrates the reconstruction
+procedure of the CMS collaboration from `raw data <http://opendata.cern.ch/search?page=1&size=20&experiment=CMS&file_type=raw>`_
+to `Analysis Object Data (AOD) <https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookDataFormats#AoD>`_,
+for data set `DoubleElectron of 2011 <http://opendata.cern.ch/record/46>`_.
+Running the commands will create ROOT files containing data in the AOD format
+specified above.
+
+The workflow consists of the steps need for the samples reconstruction, as taken
+from the `CMS legacy validation repo <https://github.com/cms-legacydata-validation/RAWToAODValidation/tree/master>`_.
+
+Reconstruction procedure
+=========================
+
+1. Input data
+--------------
+
+Any raw input data from the `CERN open data platform <http://opendata.cern.ch/search?page=1&size=20&experiment=CMS&type=Dataset&subtype=Collision&subtype=Derived&subtype=Simulated&file_type=raw>`_
+should be valid for reconstruction. In this example, the input is taken from:
+`root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/RAW/v1/000/160/433/C046161E-0D4E-E011-BCBA-0030487CD906.root`
+
+The reconstruction step can be repeated with a configuration file that depends
+on the analyzed data, e.g. `this example <http://opendata.cern.ch/record/43>`_,
+or by creating our own configuration file (created in a CMS VM) and then
+changing the script accordingly:
+
+.. code-block:: console
+
+    $ cmsDriver.py reco -s RAW2DIGI,L1Reco,RECO,USER:EventFilter/HcalRawToDigi/hcallaserhbhehffilter2012_cff.hcallLaser2012Filter --data --conditions FT_R_53_LV5::All --eventcontent AOD --customise Configuration/DataProcessing/RecoTLR.customisePrompt --no_exec --python reco_cmsdriver.py
+
+2. Compute environment
+----------------------
+In order to be able to rerun the analysis even several years in the future, we
+need to "encapsulate the current compute environment", for example to freeze the
+software package versions our analysis is using. We shall achieve this by
+preparing a `Docker <https://www.docker.com/>`_ container image for our analysis
+steps.
+
+This analysis example runs within the `CMSSW <http://cms-sw.github.io/>`_
+analysis framework that was packaged for Docker in `cmsopendata
+<https://hub.docker.com/u/cmsopendata>`_. The different images corresponds to
+data sets taken in different years. Instructions can be found under
+`this repo <http://opendata.cern.ch/docs/cms-guide-docker>`_.
+
+Moreover, the re-reconstruction task needs access run-time to the condition
+database and inside a `CMS VM <http://opendata.cern.ch/search?page=1&size=20&q=virtual%20machine&subtype=VM&type=Environment&experiment=CMS>`_,
+this is achieved with the commands:
+
+.. code-block:: console
+
+    $ ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA FT_53_LV5_AN1
+    $ ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA.db FT_53_LV5_AN1_RUNA.db
+
+For *reana*, the condition database on CVMFS can be accessed with any
+container, the only requirement is that the user should specify the necessary
+CVMFS volumes to be live-mounted in the `reana.yaml` resource section, as
+described `here <https://reana.readthedocs.io/en/latest/userguide.html#declare-necessary-resources>`_.
+
+
+3. Workflow
+-----------------
+The workflow can be logically divided into several parts:
+
+0. Upload all files.
+    Some files cannot be generated at run time and need to be uploaded.
+
+.. code-block:: console
+
+    inputs:
+      files:
+        - src/PhysicsObjectsHistos.cc
+        - BuildFile.xml
+        - demoanalyzer_cfg.py
+
+1. Fix the CMS SW environment variables manually.
+    First, we have to set up the environment variables accordingly for the
+    `CMS SW <http://cms-sw.github.io/>`_. Although this is done in the docker
+    image, `reana` overrides them and they need to be reset. This is done by
+    invoking the `cms entrypoint.sh script <https://github.com/clelange/cmssw-docker/blob/master/standalone/entrypoint.sh>`_
+    commands.
+
+    See also this `issue <https://github.com/reanahub/reana-demo-cms-reco/issues/2>`_.
+
+.. code-block:: console
+
+    $ source /opt/cms/cmsset_default.sh
+    $ scramv1 project CMSSW CMSSW_5_3_32
+    $ cd CMSSW_5_3_32/src
+    $ eval `scramv1 runtime -sh`
+
+2. Create the specific CMS path.
+    CMS specific data analysis framework requires two directory levels.
+    See also `this issue <https://github.com/reanahub/reana-demo-cms-reco/issues/8>`_.
+
+.. code-block:: console
+
+    $ mkdir Reconstruction && cd Reconstruction
+    $ mkdir Validation && cd Validation
+
+3. Create the reconstruction file.
+    See also `this repo <https://github.com/cms-legacydata-validation/RAWToAODValidation/tree/2011>`_.
+
+.. code-block:: console
+
+    $ cmsDriver.py reco -s RAW2DIGI,L1Reco,RECO,USER:EventFilter/HcalRawToDigi/hcallaserhbhehffilter2012_cff.hcallLaser2012Filter --data --filein='root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/RAW/v1/000/160/431/080D15C9-FF4D-E011-9484-0030487C7828.root' --conditions FT_53_LV5_AN1::All --eventcontent AOD --customise Configuration/DataProcessing/RecoTLR.customisePrompt --no_exec --python reco_cmsdriver.py
+
+4. Adjust the reconstruction file to the specific data file.
+    Although generated using parameters, the reconstruction file still requires
+    changes.
+
+.. code-block:: console
+
+    $ sed -i 's/from Configuration.AlCa.GlobalTag import GlobalTag/process.GlobalTag.connect = cms.string("sqlite_file:\/cvmfs\/cms-opendata-conddb.cern.ch\/FT_53_LV5_AN1_RUNA.db")/g' reco_cmsdriver.py
+    $ sed -i 's/# Other statements/from Configuration.AlCa.GlobalTag import GlobalTag/g' reco_cmsdriver.py
+    $ sed -i "s/process.GlobalTag = GlobalTag(process.GlobalTag, 'FT_53_LV5_AN1::All', '')/process.GlobalTag.globaltag = 'FT_53_LV5_AN1::All'/g" reco_cmsdriver.py
+
+5. Link the CVMFS files.
+    The `ls -l` commands are explicitly needed to make sure that the
+    `cms-opendata-conddb.cern.ch` directory has actually expanded in the image,
+    according to `this guide <http://opendata.cern.ch/docs/cms-guide-for-condition-database>`_.
+    See also `this issue <https://github.com/reanahub/reana-demo-cms-reco/issues/4>`_.
+
+.. code-block:: console
+
+    $ ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA FT_53_LV5_AN1
+    $ ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA.db FT_53_LV5_AN1_RUNA.db
+    $ ls -l
+    $ ls -l /cvmfs/
+
+6. Run the reconstruction.
+    At this point all environment variables and files should be proper.
+
+.. code-block:: console
+
+    $ cmsRun reco_cmsdriver.py
+
+7. Adjust project structure for validation
+    Copy the required files for the next steps.
+
+.. code-block:: console
+
+    $ mkdir src
+    $ scp ../../../../src/PhysicsObjectsHistos.cc ./src
+    $ scp ../../../../BuildFile.xml .
+    $ scp ../../../../demoanalyzer_cfg.py .
+
+
+8. Run CMS scram command to fix libraries.
+    Most importantly, the *BuildFile.xml* has to be inside the directory where
+    the *scram* command is executed.
+
+.. code-block:: console
+
+    $ scram b
+
+9. Run the validation file.
+    See also `this repo <http://opendata.cern.ch/record/464>`_
+
+.. code-block:: console
+
+    $ cmsRun demoanalyzer_cfg.py
+
+
+
+
+Running the example on REANA cloud
+==================================
+
+The following commands set up the *reana* environment:
+
+.. code-block:: console
+
+    $ # create new virtual environment
+    $ virtualenv ~/.virtualenvs/myreana
+    $ source ~/.virtualenvs/myreana/bin/activate
+    $ # install REANA client
+    $ pip install reana-client
+    $ # connect to some REANA cloud instance
+    $ export REANA_SERVER_URL=https://reana.cern.ch/
+    $ export REANA_ACCESS_TOKEN=XXXXXXX
+
+The workflow can be completely run using one command:
+
+.. code-block:: console
+
+    $ reana-client run
+
+It basically consists of the following steps (that can also be run
+individually):
+
+.. code-block:: console
+
+    $ # create new workflow
+    $ reana-client create -f reana.yaml
+    $ export REANA_WORKON=workflow
+    $ # start computational workflow
+    $ reana-client start
+    $ # ... should be finished in several hours, depending on the data size
+    $ reana-client status
+    $ # list workspace files
+    $ reana-client ls
+    $ # download output results
+    $ reana-client download
+
+Contributors
+============
+
+The list of contributors to this REANA example in alphabetical order:
+
+- `Daniel Prelipcean <https://orcid.org/0000-0002-4855-194X>`_
+- `Kati Lassila-Perini <https://orcid.org/0000-0002-5502-1795>`_
+- `Tibor Simko <https://orcid.org/0000-0001-7202-5803>`_

--- a/debug/reana-link-cvmfs.yaml
+++ b/debug/reana-link-cvmfs.yaml
@@ -1,0 +1,11 @@
+# Note: for this one requires the CVMFS mount.
+version: 0.6.0
+workflow:
+ type: serial
+  specification:
+    steps:
+      - environment: 'python:3.7-slim'
+        commands:
+          - ls -l /cvmfs > z1.txt
+          - find /cvmfs/cms-opendata-conddb.cern.ch > z2.txt
+

--- a/debug/reana-reco-cmsDriver.yaml
+++ b/debug/reana-reco-cmsDriver.yaml
@@ -1,0 +1,18 @@
+version: 0.6.0
+
+workflow:
+  type: serial
+  specification:
+    steps:
+      - environment: 'cmsopendata/cmssw_5_3_32'
+        commands:
+        - >
+            source /opt/cms/cmsset_default.sh
+            && scramv1 project CMSSW CMSSW_5_3_32
+            && cd CMSSW_5_3_32/src
+            && eval `scramv1 runtime -sh`
+            && cmsDriver.py reco -s RAW2DIGI,L1Reco,RECO,USER:EventFilter/HcalRawToDigi/hcallaserhbhehffilter2012_cff.hcallLaser2012Filter --data --conditions FT_R_53_LV5::All --eventcontent AOD --customise Configuration/DataProcessing/RecoTLR.customisePrompt --no_exec --python reco_cmsdriver.py
+
+outputs:
+  files:
+    - reco_cmsdriver.py

--- a/debug/reana-reco-cmsRun.yaml
+++ b/debug/reana-reco-cmsRun.yaml
@@ -1,0 +1,28 @@
+version: 0.6.0
+
+workflow:
+  type: serial
+  specification:
+    steps:
+      - environment: 'cmsopendata/cmssw_5_3_32'
+        compute_backend: htcondorcern
+        commands:
+        - >
+            source /opt/cms/cmsset_default.sh
+            && scramv1 project CMSSW CMSSW_5_3_32
+            && cd CMSSW_5_3_32/src
+            && eval `scramv1 runtime -sh`
+            && mkdir Reconstruction && cd Reconstruction
+            && mkdir Validation && cd Validation
+            && cmsDriver.py reco -s RAW2DIGI,L1Reco,RECO,USER:EventFilter/HcalRawToDigi/hcallaserhbhehffilter2012_cff.hcallLaser2012Filter --data --filein='root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/RAW/v1/000/160/431/080D15C9-FF4D-E011-9484-0030487C7828.root' --conditions FT_53_LV5_AN1::All --eventcontent AOD --no_exec --python reco_cmsdriver.py
+            && sed -i 's/from Configuration.AlCa.GlobalTag import GlobalTag/process.GlobalTag.connect = cms.string("sqlite_file:\/cvmfs\/cms-opendata-conddb.cern.ch\/FT_53_LV5_AN1_RUNA.db")/g' reco_cmsdriver.py
+            && sed -i 's/# Other statements/from Configuration.AlCa.GlobalTag import GlobalTag/g' reco_cmsdriver.py
+            && ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA FT_53_LV5_AN1
+            && ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA.db FT_53_LV5_AN1_RUNA.db
+            && ls -l
+            && ls -l /cvmfs/
+            && cmsRun reco_cmsdriver.py
+
+outputs:
+  files:
+    - reco_RAW2DIGI_L1Reco_RECO_USER.root

--- a/debug/reana-validation.yaml
+++ b/debug/reana-validation.yaml
@@ -1,0 +1,33 @@
+version: 0.6.0
+
+inputs:
+  files:
+    - demoanalyzer_cfg.py
+    - reco_RAW2DIGI_L1Reco_RECO_USER.root
+    - BuildFile.xml
+    - src/PhysicsObjectsHistos.cc
+
+workflow:
+  type: serial
+  specification:
+    steps:
+      - environment: 'cmsopendata/cmssw_5_3_32'
+        commands:
+        - >
+            source /opt/cms/cmsset_default.sh
+            && scramv1 project CMSSW CMSSW_5_3_32
+            && cd CMSSW_5_3_32/src
+            && eval `scramv1 runtime -sh`
+            && mkdir Reconstruction && cd Reconstruction
+            && mkdir Validation && cd Validation
+            && scp ../../../../demoanalyzer_cfg.py .
+            && scp ../../../reco_RAW2DIGI_L1Reco_RECO_USER.root .
+            && mkdir src
+            && scp ../../../../src/PhysicsObjectsHistos.cc ./src
+            && scp ../../../../BuildFile.xml .
+            && scram b
+            && cmsRun demoanalyzer_cfg.py
+
+outputs:
+  files:
+    - CMSSW_5_3_32/src/Reconstruction/Validation/histodemo.root

--- a/demoanalyzer_cfg.py
+++ b/demoanalyzer_cfg.py
@@ -1,0 +1,39 @@
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process("Demo")
+
+# initialize MessageLogger and output report
+process.load("FWCore.MessageLogger.MessageLogger_cfi")
+process.MessageLogger.cerr.threshold = 'INFO'
+process.MessageLogger.categories.append('Demo')
+process.MessageLogger.cerr.INFO = cms.untracked.PSet(
+    limit=cms.untracked.int32(1)
+)
+process.options = cms.untracked.PSet(
+    wantSummary=cms.untracked.bool(True),
+    SkipEvent=cms.untracked.vstring('ProductNotFound')
+)
+
+process.maxEvents = cms.untracked.PSet( input = cms.untracked.int32(20000) )
+
+process.source = cms.Source(
+    "PoolSource",
+    # replace 'myfile.root' with the source file you want to use
+    fileNames = cms.untracked.vstring('file:reco_RAW2DIGI_L1Reco_RECO_USER.root'),
+    lumisToProcess = cms.untracked.VLuminosityBlockRange('160957:1-160957:max')
+    #firstRun = cms.untracked.uint32(169957),
+    #firstEvent = cms.untracked.uint32(488034889)
+)
+
+process.demo = cms.EDAnalyzer(
+    'PhysicsObjectsHistos',
+    minTracks=cms.untracked.uint32(0)
+)
+
+process.TFileService = cms.Service(
+    "TFileService",
+    fileName=cms.string('histodemo.root')
+)
+
+
+process.p = cms.Path(process.demo)

--- a/reana-cwl.yaml
+++ b/reana-cwl.yaml
@@ -1,0 +1,7 @@
+version: 0.6.0
+inputs:
+  parameters:
+    input: workflow/input.yml
+workflow:
+  type: cwl
+  file: workflow/workflow.cwl

--- a/reana.yaml
+++ b/reana.yaml
@@ -1,0 +1,43 @@
+version: 0.6.0
+
+inputs:
+  files:
+    - src/PhysicsObjectsHistos.cc
+    - BuildFile.xml
+    - demoanalyzer_cfg.py
+
+workflow:
+  type: serial
+  specification:
+    steps:
+      - name: reco
+        environment: 'cmsopendata/cmssw_5_3_32'
+        compute_backend: kubernetes
+        commands:
+        - >
+            source /opt/cms/cmsset_default.sh
+            && scramv1 project CMSSW CMSSW_5_3_32
+            && cd CMSSW_5_3_32/src
+            && eval `scramv1 runtime -sh`
+            && mkdir Reconstruction && cd Reconstruction
+            && mkdir Validation && cd Validation
+            && cmsDriver.py reco -s RAW2DIGI,L1Reco,RECO,USER:EventFilter/HcalRawToDigi/hcallaserhbhehffilter2012_cff.hcallLaser2012Filter --data --filein='root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/RAW/v1/000/160/431/080D15C9-FF4D-E011-9484-0030487C7828.root' --conditions FT_53_LV5_AN1::All --eventcontent AOD  --no_exec --python reco_cmsdriver.py
+            && sed -i 's/from Configuration.AlCa.GlobalTag import GlobalTag/process.GlobalTag.connect = cms.string("sqlite_file:\/cvmfs\/cms-opendata-conddb.cern.ch\/FT_53_LV5_AN1_RUNA.db")/g' reco_cmsdriver.py
+            && sed -i 's/# Other statements/from Configuration.AlCa.GlobalTag import GlobalTag/g' reco_cmsdriver.py
+            && sed -i "s/process.GlobalTag = GlobalTag(process.GlobalTag, 'FT_53_LV5_AN1::All', '')/process.GlobalTag.globaltag = 'FT_53_LV5_AN1::All'/g" reco_cmsdriver.py
+            && ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA FT_53_LV5_AN1
+            && ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA.db FT_53_LV5_AN1_RUNA.db
+            && ls -l
+            && ls -l /cvmfs/
+            && cmsRun reco_cmsdriver.py
+            && mkdir src
+            && scp ../../../../src/PhysicsObjectsHistos.cc ./src
+            && scp ../../../../BuildFile.xml .
+            && scp ../../../../demoanalyzer_cfg.py .
+            && scram b
+            && cmsRun demoanalyzer_cfg.py
+
+outputs:
+  files:
+    - CMSSW_5_3_32/src/Reconstruction/Validation/reco_RAW2DIGI_L1Reco_RECO_USER.root
+    - CMSSW_5_3_32/src/Reconstruction/Validation/histodemo.root

--- a/reco_cmsdriver.py
+++ b/reco_cmsdriver.py
@@ -1,0 +1,73 @@
+# Auto generated configuration file
+# using:
+# Revision: 1.381.2.28
+# Source: /local/reps/CMSSW/CMSSW/Configuration/PyReleaseValidation/python/ConfigBuilder.py,v
+# with command line options: reco -s RAW2DIGI,L1Reco,RECO,USER:EventFilter/HcalRawToDigi/hcallaserhbhehffilter2012_cff.hcallLaser2012Filter --data --filein=root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/RAW/v1/000/160/431/080D15C9-FF4D-E011-9484-0030487C7828.root --conditions FT_53_LV5_AN1::All --eventcontent AOD --no_exec --python reco_cmsdriver.py
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process('RECO')
+
+# import of standard configurations
+process.load('Configuration.StandardSequences.Services_cff')
+process.load('SimGeneral.HepPDTESSource.pythiapdt_cfi')
+process.load('FWCore.MessageService.MessageLogger_cfi')
+process.load('Configuration.EventContent.EventContent_cff')
+process.load('Configuration.StandardSequences.GeometryRecoDB_cff')
+process.load('Configuration.StandardSequences.MagneticField_AutoFromDBCurrent_cff')
+process.load('Configuration.StandardSequences.RawToDigi_Data_cff')
+process.load('Configuration.StandardSequences.L1Reco_cff')
+process.load('Configuration.StandardSequences.Reconstruction_Data_cff')
+process.load('EventFilter.HcalRawToDigi.hcallaserhbhehffilter2012_cff')
+process.load('Configuration.StandardSequences.EndOfProcess_cff')
+process.load('Configuration.StandardSequences.FrontierConditions_GlobalTag_cff')
+
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(1)
+)
+
+# Input source
+process.source = cms.Source("PoolSource",
+    secondaryFileNames = cms.untracked.vstring(),
+    fileNames = cms.untracked.vstring('root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/RAW/v1/000/160/431/080D15C9-FF4D-E011-9484-0030487C7828.root')
+)
+
+process.options = cms.untracked.PSet(
+
+)
+
+# Production Info
+process.configurationMetadata = cms.untracked.PSet(
+    version = cms.untracked.string('$Revision: 1.381.2.28 $'),
+    annotation = cms.untracked.string('reco nevts:1'),
+    name = cms.untracked.string('PyReleaseValidation')
+)
+
+# Output definition
+
+process.AODoutput = cms.OutputModule("PoolOutputModule",
+    eventAutoFlushCompressedSize = cms.untracked.int32(15728640),
+    outputCommands = process.AODEventContent.outputCommands,
+    fileName = cms.untracked.string('reco_RAW2DIGI_L1Reco_RECO_USER.root'),
+    dataset = cms.untracked.PSet(
+        filterName = cms.untracked.string(''),
+        dataTier = cms.untracked.string('')
+    )
+)
+
+# Additional output definition
+
+from Configuration.AlCa.GlobalTag import GlobalTag
+process.GlobalTag.connect = cms.string("sqlite_file:/cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA.db")
+process.GlobalTag = GlobalTag(process.GlobalTag, 'FT_53_LV5_AN1::All', '')
+
+# Path and EndPath definitions
+process.raw2digi_step = cms.Path(process.RawToDigi)
+process.L1Reco_step = cms.Path(process.L1Reco)
+process.reconstruction_step = cms.Path(process.reconstruction)
+process.user_step = cms.Path(process.hcallLaser2012Filter)
+process.endjob_step = cms.EndPath(process.endOfProcess)
+process.AODoutput_step = cms.EndPath(process.AODoutput)
+
+# Schedule definition
+process.schedule = cms.Schedule(process.raw2digi_step,process.L1Reco_step,process.reconstruction_step,process.user_step,process.endjob_step,process.AODoutput_step)
+

--- a/src/PhysicsObjectsHistos.cc
+++ b/src/PhysicsObjectsHistos.cc
@@ -1,0 +1,457 @@
+// -*- C++ -*-
+//
+// Class:      PhysicsObjectsHistos
+// 
+/**\class PhysicsObjectsHistos PhysicsObjectsHistos.cc 
+
+*/
+//
+// $Id$
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "TH1.h"
+
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+
+//yop electrones
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectronFwd.h"
+
+//yop fotones
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+#include "DataFormats/EgammaCandidates/interface/PhotonFwd.h"
+
+//yop jets
+#include "DataFormats/JetReco/interface/PFJet.h"
+
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+
+//yop taus
+#include "DataFormats/TauReco/interface/PFTauDiscriminator.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+
+//missing energy
+#include "DataFormats/METReco/interface/MET.h"
+#include "DataFormats/METReco/interface/METCollection.h"
+#include "DataFormats/METReco/interface/METFwd.h"
+
+#include "DataFormats/METReco/interface/PFMET.h"
+#include "DataFormats/METReco/interface/PFMETCollection.h"
+#include "DataFormats/METReco/interface/PFMETFwd.h"
+
+
+
+//
+// class decleration
+//
+
+class PhysicsObjectsHistos : public edm::EDAnalyzer {
+public:
+  explicit PhysicsObjectsHistos(const edm::ParameterSet&);
+  ~PhysicsObjectsHistos();
+
+
+private:
+  virtual void beginJob() ;
+  virtual void analyze(const edm::Event&, const edm::EventSetup&);
+  virtual void endJob() ;
+
+  // ----------member data ---------------------------
+  unsigned int minTracks_;
+  TH1D *demohisto;
+TH1D *h1;
+TH1D *h2;
+TH1D *h3;
+TH1D *h4;
+
+TH1D *h11;
+TH1D *h12;
+TH1D *h13;
+TH1D *h14;
+
+  //yop histogramas elec
+TH1D *h21;
+TH1D *h22;
+TH1D *h23;
+TH1D *h24;
+
+  //histogramas photons
+TH1D *h31;
+TH1D *h32;
+TH1D *h33;
+TH1D *h34;
+
+  //histogram jets
+TH1D *h41;
+TH1D *h42;
+TH1D *h43;
+TH1D *h44;
+
+  //histograms taus
+TH1D *h51;
+TH1D *h52;
+TH1D *h53;
+TH1D *h54;
+
+  //missing transverse energy
+TH1D *h61;
+TH1D *h62;
+TH1D *h63;
+TH1D *h64;
+
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+
+PhysicsObjectsHistos::PhysicsObjectsHistos(const edm::ParameterSet& iConfig) :
+  minTracks_(iConfig.getUntrackedParameter<unsigned int>("minTracks",0))
+{
+
+  //now do what ever initialization is needed
+
+  // I want to make a histogram of number of tracks in an event
+     
+  edm::Service<TFileService> fs;
+  demohisto = fs->make<TH1D>("tracks" , "Tracks" , 100 , 0 , 5000 );
+
+h1 = fs->make<TH1D>("TR_momentum", "Track momentum", 100, 0., 20.); 
+h1->GetXaxis()->SetTitle("Track Momentum (in GeV/c)");
+h1->GetYaxis()->SetTitle("Number of Events");
+
+// track Transverse_momentum
+h2 = fs->make<TH1D>("TR_Transverse_momentum", "Track transverse momentum", 100, 0., 20.); 
+h2->GetXaxis()->SetTitle("Transverse Momentum of tracks (in GeV/c)");
+h2->GetYaxis()->SetTitle("Number of Events");
+
+// track pseudorapity
+h3 = fs->make<TH1D>("TR_eta", "Track eta", 140, -3.5, 3.5);
+h3->GetXaxis()->SetTitle("Eta of tracks");
+h3->GetYaxis()->SetTitle("Number of Events");
+
+// global muon azimuth angle  
+h4 = fs->make<TH1D>("TR_phi", "Track phi", 314, -3.15, 3.15);
+h4->GetXaxis()->SetTitle("Phi (in radians)");
+h4->GetYaxis()->SetTitle("Number of Events");
+
+h11 = fs->make<TH1D>("Mu_momentum", "Muon momentum", 100, 0., 20.); 
+h11->GetXaxis()->SetTitle("Muon Momentum (in GeV/c)");
+h11->GetYaxis()->SetTitle("Number of Events");
+
+//  muon Transverse_momentum
+h12 = fs->make<TH1D>("Mu_Transverse_momentum", "Muon transverse momentum", 100, 0., 20.); 
+h12->GetXaxis()->SetTitle("Transverse Momentum of muons (in GeV/c)");
+h12->GetYaxis()->SetTitle("Number of Events");
+
+//  muon pseudorapity
+h13 = fs->make<TH1D>("Mu_eta", "Muon eta", 140, -3.5, 3.5);
+h13->GetXaxis()->SetTitle("Eta of muons");
+h13->GetYaxis()->SetTitle("Number of Events");
+
+//  muon azimuth angle  
+h14 = fs->make<TH1D>("Mu_phi", "M_phi", 314, -3.15, 3.15);
+h14->GetXaxis()->SetTitle("Phi (in radians)");
+h14->GetYaxis()->SetTitle("Number of Events");
+
+//electrones his
+
+h21 = fs->make<TH1D>("Ele_momentum", "Electron momentum", 100, 0., 40.); 
+h21->GetXaxis()->SetTitle("Electron Momentum (in GeV/c)");
+h21->GetYaxis()->SetTitle("Number of Events");
+
+//  electron Transverse_momentum
+h22 = fs->make<TH1D>("Ele_Transverse_momentum", "Electron transverse momentum", 100, 0., 40.); 
+h22->GetXaxis()->SetTitle("Transverse Momentum of electrons (in GeV/c)");
+h22->GetYaxis()->SetTitle("Number of Events");
+
+//  electon pseudorapity
+h23 = fs->make<TH1D>("Ele_eta", "Electron eta", 140, -3.5, 3.5);
+h23->GetXaxis()->SetTitle("Eta of electrons");
+h23->GetYaxis()->SetTitle("Number of Events");
+
+//  electron azimuth angle  
+h24 = fs->make<TH1D>("Ele_phi", "Electron phi", 314, -3.15, 3.15);
+h24->GetXaxis()->SetTitle("Phi (in radians)");
+h24->GetYaxis()->SetTitle("Number of Events");
+
+//photons
+
+//photon his
+
+h31 = fs->make<TH1D>("Gam_momentum", "Photon momentum", 100, 0., 40.); 
+h31->GetXaxis()->SetTitle("Photon Momentum (in GeV/c)");
+h31->GetYaxis()->SetTitle("Number of Events");
+
+//  photon Transverse_momentum
+h32 = fs->make<TH1D>("Gam_Transverse_momentum", "Photon transverse momentum", 100, 0., 40.); 
+h32->GetXaxis()->SetTitle("Transverse Momentum of photons (in GeV/c)");
+h32->GetYaxis()->SetTitle("Number of Events");
+
+//  photon  pseudorapity
+h33 = fs->make<TH1D>("Gam_eta", "Photon eta", 140, -3.5, 3.5);
+h33->GetXaxis()->SetTitle("Eta of photons");
+h33->GetYaxis()->SetTitle("Number of Events");
+
+//  photon  azimuth angle  
+h34 = fs->make<TH1D>("Gam_phi", "Photon phi", 314, -3.15, 3.15);
+h34->GetXaxis()->SetTitle("Phi (in radians)");
+h34->GetYaxis()->SetTitle("Number of Events");
+
+
+//jets
+
+//jet his
+
+h41 = fs->make<TH1D>("Jet_momentum", "Jet momentum", 100, 0., 40.); 
+h41->GetXaxis()->SetTitle("Jet Momentum (in GeV/c)");
+h41->GetYaxis()->SetTitle("Number of Events");
+
+//  jet Transverse_momentum
+h42 = fs->make<TH1D>("Jet_Transverse_momentum", "Jet transverse momentum", 100, 0., 40.); 
+h42->GetXaxis()->SetTitle("Transverse Momentum of jets (in GeV/c)");
+h42->GetYaxis()->SetTitle("Number of Events");
+
+//  jet  pseudorapity
+h43 = fs->make<TH1D>("Jet_eta", "Jet eta", 140, -3.5, 3.5);
+h43->GetXaxis()->SetTitle("Eta of jets");
+h43->GetYaxis()->SetTitle("Number of Events");
+
+//  jet  azimuth angle  
+h44 = fs->make<TH1D>("Jet_phi", "Jet phi", 314, -3.15, 3.15);
+h44->GetXaxis()->SetTitle("Phi (in radians)");
+h44->GetYaxis()->SetTitle("Number of Events");
+
+//taus
+
+// his
+
+h51 = fs->make<TH1D>("Tau_momentum", "Tau momentum", 100, 0., 20.); 
+h51->GetXaxis()->SetTitle("Tau Momentum (in GeV/c)");
+h51->GetYaxis()->SetTitle("Number of Events");
+
+//  Transverse_momentum
+h52 = fs->make<TH1D>("Tau_Transverse_momentum", "Tau transverse momentum", 100, 0., 20.); 
+h52->GetXaxis()->SetTitle("Transverse Momentum of taus (in GeV/c)");
+h52->GetYaxis()->SetTitle("Number of Events");
+
+//   pseudorapity
+h53 = fs->make<TH1D>("Tau_eta", "Tau eta", 140, -3.5, 3.5);
+h53->GetXaxis()->SetTitle("Eta of taus");
+h53->GetYaxis()->SetTitle("Number of Events");
+
+//  azimuth angle  
+h54 = fs->make<TH1D>("Tau_phi", "Tau phi", 314, -3.15, 3.15);
+h54->GetXaxis()->SetTitle("Phi (in radians)");
+h54->GetYaxis()->SetTitle("Number of Events");
+
+//MET 
+
+// his
+
+h61 = fs->make<TH1D>("MET_sumet", "MET Sum Et", 100, 0., 20.); 
+h61->GetXaxis()->SetTitle("MET Sum Et (in GeV/c)");
+h61->GetYaxis()->SetTitle("Number of Events");
+
+//  Transverse_momentum
+h62 = fs->make<TH1D>("MET_Transverse_momentum", "MET transverse momentum", 100, 0., 20.); 
+h62->GetXaxis()->SetTitle("Transverse Momentum of MET (in GeV/c)");
+h62->GetYaxis()->SetTitle("Number of Events");
+
+//   pseudorapity
+h63 = fs->make<TH1D>("MET_eta", "MET eta", 140, -3.5, 3.5);
+h63->GetXaxis()->SetTitle("Eta of MET");
+h63->GetYaxis()->SetTitle("Number of Events");
+
+//  azimuth angle  
+h64 = fs->make<TH1D>("MET_phi", "MET phi", 314, -3.15, 3.15);
+h64->GetXaxis()->SetTitle("Phi (in radians)");
+h64->GetYaxis()->SetTitle("Number of Events");
+
+
+
+
+
+}
+
+
+PhysicsObjectsHistos::~PhysicsObjectsHistos()
+{
+ 
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called to for each event  ------------
+void
+PhysicsObjectsHistos::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  using namespace edm;
+
+   Handle<reco::TrackCollection> tracks;
+   iEvent.getByLabel("generalTracks", tracks); 
+   //   LogInfo("Demo") << "number of tracks "<<tracks->size();
+   demohisto->Fill(tracks->size());
+
+// WHAT: Loop over all the Global Muons of current Event
+// WHY:  to select good candidates to be used in invariant mass calculation
+  for (reco::TrackCollection::const_iterator it = tracks->begin(); 
+     it != tracks->end(); it++) {
+     h1->Fill(it->p());
+     h2->Fill(it->pt());
+     h3->Fill(it->eta());
+     h4->Fill(it->phi());
+
+   }
+
+   Handle<reco::MuonCollection> mymuons;
+   iEvent.getByLabel("muons", mymuons); 
+     for(reco::MuonCollection::const_iterator itMuon = mymuons->begin();
+       itMuon != mymuons->end();
+       ++itMuon) {
+         h11->Fill(itMuon->p());
+    h12->Fill(itMuon->pt());
+    h13->Fill(itMuon->eta());
+    h14->Fill(itMuon->phi());//  your code
+   }
+
+     //electrones
+   Handle<reco::GsfElectronCollection> electrons;
+   iEvent.getByLabel("gsfElectrons", electrons); 
+   for(reco::GsfElectronCollection::const_iterator itElectron = electrons->begin();
+       itElectron != electrons->end();
+       ++itElectron) {
+    h21->Fill(itElectron->p());
+    h22->Fill(itElectron->pt());
+    h23->Fill(itElectron->eta());
+    h24->Fill(itElectron->phi());
+   }
+
+   //PHOTONS
+ Handle<reco::PhotonCollection> photons;
+   iEvent.getByLabel("photons", photons); 
+   // LogInfo("Demo") << "number of photons "<<photons->size();  
+for(reco::PhotonCollection::const_iterator itphotons = photons->begin();
+       itphotons != photons->end();
+       ++itphotons) {
+    h31->Fill(itphotons->p());
+    h32->Fill(itphotons->pt());
+    h33->Fill(itphotons->eta());
+    h34->Fill(itphotons->phi());
+   }
+
+  //jets
+ Handle<reco::PFJetCollection> ak5PFJets;
+   iEvent.getByLabel("ak5PFJets", ak5PFJets); 
+   for(reco::PFJetCollection::const_iterator itjets = ak5PFJets->begin();
+       itjets != ak5PFJets->end();
+       ++itjets) {
+    h41->Fill(itjets->p());
+    h42->Fill(itjets->pt());
+    h43->Fill(itjets->eta());
+    h44->Fill(itjets->phi());
+   }
+
+   //taus
+   Handle<reco::PFTauCollection> taus;
+   iEvent.getByLabel("hpsPFTauProducer", taus); 
+   Handle<reco::PFTauDiscriminator> discr;
+   iEvent.getByLabel("hpsPFTauDiscriminationByDecayModeFinding", discr); 
+   for ( unsigned iTau = 0; iTau < taus->size(); ++iTau ) {
+        reco::PFTauRef tauCandidate(taus, iTau);
+// check if tau candidate has passed discriminator
+        if( (*discr)[tauCandidate] > 0.5 ){
+	  h51->Fill((*tauCandidate).p());
+	  h52->Fill((*tauCandidate).pt());
+	  h53->Fill((*tauCandidate).eta());
+	  h54->Fill((*tauCandidate).phi());
+        }
+    }
+   //for(reco::PFTauCollection::const_iterator itTaus = taus->begin();
+
+   //    itTaus != taus->end();
+   //    ++itTaus) {
+   // h51->Fill(itTaus->p());
+   // h52->Fill(itTaus->pt());
+   // h53->Fill(itTaus->eta());
+   //  h54->Fill(itTaus->phi());
+   //}
+
+   // MET
+   Handle<reco::PFMETCollection> pfMet;
+   iEvent.getByLabel("pfMet", pfMet); 
+   // for(reco::PFMETCollection::const_iterator itmet = pfMet->begin();
+   //  itmet != pfMet->end();
+   //  ++itmet) {
+
+  h61->Fill((*pfMet)[0].sumEt());
+  h62->Fill((*pfMet)[0].et());
+  h63->Fill((*pfMet)[0].eta());
+  h64->Fill((*pfMet)[0].phi());
+   // }
+
+
+
+
+#ifdef THIS_IS_AN_EVENT_EXAMPLE
+  Handle<ExampleData> pIn;
+  iEvent.getByLabel("example",pIn);
+#endif
+   
+#ifdef THIS_IS_AN_EVENTSETUP_EXAMPLE
+  ESHandle<SetupData> pSetup;
+  iSetup.get<SetupRecord>().get(pSetup);
+#endif
+}
+
+
+// ------------ method called once each job just before starting event loop  ------------
+void 
+PhysicsObjectsHistos::beginJob()
+{
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void 
+PhysicsObjectsHistos::endJob() {
+}
+
+//define this as a plug-in
+DEFINE_FWK_MODULE(PhysicsObjectsHistos);

--- a/workflow/input.yml
+++ b/workflow/input.yml
@@ -1,0 +1,9 @@
+library:
+  class: File
+  path: src/PhysicsObjectsHistos.cc
+build_file:
+  class: File
+  path: BuildFile.xml
+validation_script:
+  class: File
+  path: demoanalyzer_cfg.py

--- a/workflow/reco.cwl
+++ b/workflow/reco.cwl
@@ -1,0 +1,54 @@
+class: CommandLineTool
+cwlVersion: v1.0
+
+requirements:
+  - class: DockerRequirement
+    dockerPull: cmsopendata/cmssw_5_3_32
+
+baseCommand:
+  - /bin/zsh
+
+inputs:
+  library: File
+  build_file: File
+  validation_script: File
+
+arguments:
+  - position: 0
+    prefix: '-c'
+    valueFrom: |
+      source /opt/cms/cmsset_default.sh
+      scramv1 project CMSSW CMSSW_5_3_32
+      cd CMSSW_5_3_32/src
+      eval `scramv1 runtime -sh`
+      mkdir Reconstruction && cd Reconstruction
+      mkdir Validation && cd Validation
+      cmsDriver.py reco -s RAW2DIGI,L1Reco,RECO,USER:EventFilter/HcalRawToDigi/hcallaserhbhehffilter2012_cff.hcallLaser2012Filter --data --filein='root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/RAW/v1/000/160/431/080D15C9-FF4D-E011-9484-0030487C7828.root' --conditions FT_53_LV5_AN1::All --eventcontent AOD  --no_exec --python reco_cmsdriver.py
+      sed -i 's/from Configuration.AlCa.GlobalTag import GlobalTag/process.GlobalTag.connect = cms.string("sqlite_file:\/cvmfs\/cms-opendata-conddb.cern.ch\/FT_53_LV5_AN1_RUNA.db")/g' reco_cmsdriver.py
+      sed -i 's/# Other statements/from Configuration.AlCa.GlobalTag import GlobalTag/g' reco_cmsdriver.py
+      sed -i "s/process.GlobalTag = GlobalTag(process.GlobalTag, 'FT_53_LV5_AN1::All', '')/process.GlobalTag.globaltag = 'FT_53_LV5_AN1::All'/g" reco_cmsdriver.py
+      ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA FT_53_LV5_AN1
+      ln -sf /cvmfs/cms-opendata-conddb.cern.ch/FT_53_LV5_AN1_RUNA.db FT_53_LV5_AN1_RUNA.db
+      ls -l
+      ls -l /cvmfs/
+      cmsRun reco_cmsdriver.py
+      mkdir src
+      scp ../../../../../../src/$(inputs.library.basename) ./src
+      scp ../../../../../../$(inputs.build_file.basename) .
+      scp ../../../../../../$(inputs.validation_script.basename) .
+      scram b
+      cmsRun $(inputs.validation_script.basename)
+
+outputs:
+  - id: result.root
+    type: File
+    outputBinding:
+      glob: CMSSW_5_3_32/src/Reconstruction/Validation/reco_RAW2DIGI_L1Reco_RECO_USER.root
+  - id: histo.root
+    type: File
+    outputBinding:
+      glob: CMSSW_5_3_32/src/Reconstruction/Validation/histodemo.root
+  - id: reco.log
+    type: stdout
+
+stdout: reco.log

--- a/workflow/workflow.cwl
+++ b/workflow/workflow.cwl
@@ -1,0 +1,48 @@
+#!/usr/bin/env cwl-runner
+
+# Note that if you are working on the analysis development locally, i.e. outside
+# of the REANA platform, you can proceed as follows:
+#
+#   $ mkdir cwl-local-run
+#   $ cd cwl-local-run
+#   $ cp -a ../workflow/input.yml .
+#   $ cp -a ../BuildFile.xml ../demoanalyzer_cfg.py .
+#   $ mkdir src
+#   $ cp -a ../src/PhysicsObjectsHistos.cc ./src/
+#   $ cwltool --quiet --outdir="../results" ../workflow/workflow.cwl input.yml
+
+
+cwlVersion: v1.0
+class: Workflow
+
+inputs:
+  library:
+    type: File
+  build_file:
+    type: File
+  validation_script:
+    type: File
+
+outputs:
+  result.root:
+    type: File
+    outputSource:
+      reco/result.root
+  histo.root:
+    type: File
+    outputSource:
+      reco/histo.root
+  reco.log:
+    type: File
+    outputSource:
+      reco/reco.log
+
+
+steps:
+  reco:
+    in:
+      library: library
+      build_file: build_file
+      validation_script: validation_script
+    run: reco.cwl
+    out: [result.root, histo.root, reco.log]

